src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -326,12 +329,145 @@ def update_start_times(db_manager: DatabaseManager) -> None:
         )
 
 
+def update_branch_trunk_route_id(db_manager: DatabaseManager) -> None:
+    """
+    update `branch_route_id` and `trunk_route_id` fields in trips table
+    """
+    distinct_t_trips = (
+        sa.select(TempEventCompare.service_date, TempEventCompare.pm_trip_id)
+        .distinct()
+        .subquery("distinct_trips")
+    )
+
+    distinct_trips = (
+        sa.select(
+            VehicleTrips.pm_trip_id,
+            VehicleTrips.route_id,
+            VehicleTrips.branch_route_id,
+            VehicleTrips.trunk_route_id,
+        )
+        .select_from(VehicleTrips)
+        .join(
+            distinct_t_trips,
+            sa.and_(
+                distinct_t_trips.c.service_date == VehicleTrips.service_date,
+                distinct_t_trips.c.pm_trip_id == VehicleTrips.pm_trip_id,
+            ),
+        )
+    )
+
+    distinct_red_trips = (
+        sa.select(TempEventCompare.service_date, TempEventCompare.pm_trip_id)
+        .distinct()
+        .where(TempEventCompare.route_id == "Red")
+        .subquery("distinct_trips")
+    )
+
+    red_events = (
+        sa.select(
+            VehicleEvents.pm_trip_id,
+            VehicleEvents.stop_id,
+        )
+        .select_from(VehicleEvents)
+        .join(
+            distinct_red_trips,
+            sa.and_(
+                distinct_red_trips.c.service_date == VehicleEvents.service_date,
+                distinct_red_trips.c.pm_trip_id == VehicleEvents.pm_trip_id,
+            ),
+        )
+    )
+
+    process_logger = ProcessLogger("l1_trips.update_branch_trunk_route_id")
+    process_logger.log_start()
+
+    trips_df = db_manager.select_as_dataframe(distinct_trips)
+    red_events_df = db_manager.select_as_dataframe(red_events)
+
+    def get_red_branch(pm_trip_id: int) -> Optional[str]:
+        ashmont_stop_ids = {
+            "70087",
+            "70088",
+            "70089",
+            "70090",
+            "70091",
+            "70092",
+            "70093",
+            "70094",
+            "70085",
+            "70086",
+        }
+        braintree_stop_ids = {
+            "70097",
+            "70098",
+            "70099",
+            "70100",
+            "70101",
+            "70102",
+            "70103",
+            "70104",
+            "70105",
+            "70095",
+            "70096",
+        }
```

Comment:
i have an intuition that these values should be stored in a different place, but i think thats more something to think about if / when we convert from DB to Flat File architecture.